packages/flutter/lib/src/widgets/scroll_view.dart

```diff
@@ -21,14 +21,15 @@ abstract class ScrollView extends StatelessWidget {
     this.scrollDirection: Axis.vertical,
     this.reverse: false,
     this.controller,
-    this.primary: false,
+    bool primary,
     this.physics,
     this.shrinkWrap: false,
-  }) : super(key: key) {
-    assert(reverse != null);
-    assert(shrinkWrap != null);
-    assert(primary != null);
-    assert(controller == null || !primary,
+  }) : primary = primary ?? controller == null && scrollDirection == Axis.vertical,
+       super(key: key) {
+    assert(this.reverse != null);
+    assert(this.shrinkWrap != null);
+    assert(this.primary != null);
+    assert(this.controller == null || !this.primary,
            'Primary ScrollViews obtain their ScrollController via inheritance from a PrimaryScrollController widget. '
            'You cannot both set primary to true and pass an explicit controller.'
     );
```

Comment:
Since this is subtle and it's in our minds right now, please document the semantics of `primary`, with particular emphasis on its default value, as a dartdoc on the `primary` field below.